src/streams/mod.rs

```diff
@@ -290,21 +303,16 @@ impl Streams {
 
         if let Some(rtx) = rtx {
             stream.maybe_reset_rtx(rtx);
-            associate_ssrc_mid(&mut self.source_keys_rx, rtx, mid, ssrc, reason);
         }
-        associate_ssrc_mid(&mut self.source_keys_rx, ssrc, mid, ssrc, reason);
 
         stream
     }
 
     pub fn remove_stream_rx(&mut self, ssrc: Ssrc) -> bool {
         let stream = self.streams_rx.remove(&ssrc);
         let existed = stream.is_some();
-        if let Some(rtx) = stream.and_then(|s| s.rtx()) {
-            self.source_keys_rx.remove(&rtx);
-        }
 
-        self.source_keys_rx.remove(&ssrc);
+        self.rx_lookup.retain(|k, l| *k != ssrc && l.main != ssrc);
```

Comment:
Should this `&&` not be `||` to also remove rtx streams?

Reply:
`retain` is inverse logic no?

Reply:
Right... Not enough coffee ☕